### PR TITLE
Fix tests for Basic JavaScript: Write Reusable JavaScript with Functions

### PR DESCRIPTION
**Pull request by T1mL3arn**
- fix original log() method restoration
- fix case that other userscripts(or extensions) can affect console's output and therefore can fail the tests

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

### CODE REVIEW

**Comment by RandellDawson** [comment]
@T1mL3arn Just so we are clear on your intent, what is an example of a case where "other userscripts(or extensions) can affect console output and therefore can ruin the test"?

**Comment by T1mL3arn** [comment]
@RandellDawson This is what I think it was:

Any userscript or browser's extension can log info to console. Sometimes this happens just before tests, therefore even if the user's print is correct the tests gets an userscript's output and fails.

That happened to me - I have some scripts which log messages to console and I was not able to pass this challenge. But after disabling all problem scripts problem was gone.

So now 
```js
if(message === 'Hi World')  hiWorldWasLogged = true;
assert(hiWorldWasLogged, '<code>reusableFunction</code> should output "Hi World" to the dev console');
```
ensures that user's function prints what is expected and tests just check this flag.

I can try to write minimal example, if that explanation is not enough.

**Comment by lynxlynxlynx** [comment]
LGTM, but needs another local tester.

**Comment by ojeytonwilliams** [comment]
LGTM too except that `logOutput' should be removed since it is no longer used in the tests.  It does turn up in the teardown, but I think the teardown should just be 
```js
uncapture();
```
as the two IIFEs do nothing, as far as I can see.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/write-reusable-javascript-with-functions.english.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ tests:
   - text: <code>reusableFunction</code> should be a function
     testString: assert(typeof reusableFunction === 'function', '<code>reusableFunction</code> should be a function');
   - text: <code>reusableFunction</code> should output "Hi World" to the dev console
-    testString: assert("Hi World" === logOutput, '<code>reusableFunction</code> should output "Hi World" to the dev console');
+    testString: assert(hiWorldWasLogged, '<code>reusableFunction</code> should output "Hi World" to the dev console');
   - text: Call <code>reusableFunction</code> after you define it
     testString: assert(/^\s*reusableFunction\(\)\s*;/m.test(code), 'Call <code>reusableFunction</code> after you define it');
```
Comment (thecodingaviator) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/write-reusable-javascript-with-functions.english.md]:
```suggestion
    testString: assert(/^\s*reusableFunction\(\)\s*/m.test(code), 'Call <code>reusableFunction</code> after you define it');
```
The function can be called without the `;`

**Review by thecodingaviator — CHANGES_REQUESTED** [review]
Check comment

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/write-reusable-javascript-with-functions.english.md (hunk 1 above)
Comment (lynxlynxlynx) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/write-reusable-javascript-with-functions.english.md]: But all the others are using it, so why introduce an inconsistency?

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/write-reusable-javascript-with-functions.english.md]: @lynxlynxlynx Not all the others.  On many occasions we have been allowing the user to not include the semi-colon (where it is applicable not to). In the examples we show "best practice", but the tests should allow valid syntax.

Comment (lynxlynxlynx) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/write-reusable-javascript-with-functions.english.md]: All others in this file, of course. However I see I misread the diff in the email, so forget I mentioned anything.

**Review by thecodingaviator — APPROVED** [review]
LGTM! 👍

**Comment by thecodingaviator** [comment]
@T1mL3arn Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**